packages/rocketchat-theme/server/colors.less

```diff
@@ -155,8 +155,14 @@
 	background-color: @transparent-darker;
 }
 
-.background-transparent-darker-hover:hover {
-	background-color: @transparent-darker;
+@media all and(min-width: 781px) {
+	.background-transparent-darker-hover:hover {
```

Comment:
You can't limit this global classes to desktop only, you need remove or add this class with JS when you target specific screen sizes.